src/database/connection.py

```
@@ -52,6 +70,127 @@ def __init__(self, minconn: int, maxconn: int, **conn_kwargs: Any):
+        # Perform health check if needed
+        if not self.check_health():
+            logger.warning("Database health check failed during connection validation")
+            return False
+
+        # Try a simple query to validate connection
+        try:
+            with conn.cursor() as cur:
+                cur.execute("SELECT 1")
+                result = cur.fetchone()
+                return result is not None and result[0] == 1
+        except Exception as exc:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)